File: app/src/main/java/za/co/kernelpanic/cloudy/di/modules/WeatherActivityModule.java
package za.co.kernelpanic.cloudy.di.modules;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;
import za.co.kernelpanic.cloudy.ui.activities.WeatherActivity;
import za.co.kernelpanic.cloudy.di.submodules.FragmentBuildersModule;

@Module
public abstract class WeatherActivityModule {

    /*
     * by
     */
    @ContributesAndroidInjector (modules = {FragmentBuildersModule.class})
    abstract WeatherActivity contributesWeatherActivity();
}

File: app/src/main/java/za/co/kernelpanic/cloudy/di/modules/WeatherActivityModule.kt
package za.co.kernelpanic.cloudy.di.modules

import dagger.Module
import dagger.android.ContributesAndroidInjector
import za.co.kernelpanic.cloudy.ui.activities.WeatherActivity
import za.co.kernelpanic.cloudy.di.submodules.FragmentBuildersModule

@Module
abstract class WeatherActivityModule {

    @ContributesAndroidInjector(modules = [FragmentBuildersModule::class])
    abstract fun contributesWeatherActivity(): WeatherActivity
}
